feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,619 @@
+	for _, p4Client := range p4Clients {
+		removeClient(p4Client)
+	}
+}
+
+func TestReplacePrimary(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []mcTestArgs{
+		{
+			name:       "Primary_secondary_OK",
+			expectPass: true,
+			Items: [2]*testArgs{
+				{
+					desc:              "Primary connection",
+					lowID:             *ygot.Uint64(101),
+					highID:            *ygot.Uint64(0),
+					handle:            clientConnection(t, dut),
+					deviceID:          deviceID,
+					expectPass:        true,
+					expectWrite:       true,
+					expectRead:        true,
+					expectStatus:      0,
+					expectFinalStatus: 6,
+				},
+				{
+					desc:              "new Primary",
+					lowID:             *ygot.Uint64(102),
+					highID:            *ygot.Uint64(0),
+					handle:            clientConnection(t, dut),
+					deviceID:          deviceID,
+					expectPass:        true,
+					expectWrite:       true,
+					expectRead:        true,
+					expectStatus:      0,
+					expectFinalStatus: 0,
+				},
+			},
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			resp, err := streamP4RTArb(test.Items[0])
+			if err != nil && test.expectPass {
+				t.Errorf("Failed to setup P4RT Client: %v", err)
+			}
+			// Validate status code
+			if resp != test.Items[0].expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.Items[0].expectStatus, resp)
+			}
+			// Validates that client0 can read/write
+			validateRWResp(t, test.Items[0])
+			// Create the stream for client1 (new client)
+			newResp, err := streamP4RTArb(test.Items[1])
+			if err != nil {
+				t.Errorf("Failed to setup P4RT Client: %v", err)
+			}
+			if newResp != test.Items[1].expectStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.Items[1].expectStatus, newResp)
+			}
+			// Validates that client1 can read/write
+			validateRWResp(t, test.Items[1])
+			time.Sleep(1 * time.Second)
+			// get first client new response code
+			arbResp, err := getRespCode(test.Items[0])
+			if err != nil {
+				t.Errorf("Failed get response code from primary: %v", err)
+			}
+			if arbResp != test.Items[0].expectFinalStatus {
+				t.Errorf("Incorrect status code received: want %d, got %d", test.Items[0].expectFinalStatus, arbResp)
+			}
+			// Since client1 has connected with higher election ID
+			// the expectWrite flag is moved to false
+			test.Items[0].expectWrite = false
+			// Validates that client0 can only read, cannot write anymore
+			validateRWResp(t, test.Items[0])
+		})
+		// Teardown clients
+		for _, item := range test.Items {
+			item.handle.StreamChannelDestroy(&streamName)
+			item.handle.ServerDisconnect()
+		}
+	}
+}
+
+// Test arbitration update from same client
+func TestArbitrationUpdate(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	configureDeviceId(t, dut)
+	configurePortId(t, dut)
+	testCases := []testArgs{
```

Comment:
Single testArg. Doesn't need a table driven test and the loop over testCases.

Reply:
I can do that, however if there are other tests to be added in the future we'll have to re-implement table driven test  + loop.